.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,47 @@
+name: Build and deploy
+
+on: [push]
+
+jobs:
+  Deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3.4.1
+
+      - name: Install Pnpm
+        run: npm install -g pnpm
+
+      - name: Build frontend
+        run: |
+
+          cd frontend
+          pnpm install
+          pnpm run build
+
+      - name: Check the folder
+        run: |
+
+          cd frontend
+          ls
```

Comment:
This `ls` block was only used by you for debugging, it's not necessary in the end result.